### PR TITLE
Simplified bundle directory

### PR DESCRIPTION
**Pull request by fulmicoton**
This PR simplifies the bundle directory.

This was required for the merge PR, as it avoids copying the data fetched into a `Vec<u8>`.
In the merge PR, we use this directory with a file slice backed by a Mmap, so that we entirely avoid the copy
(and the anonymous memory).

### CODE REVIEW

File: quickwit-storage/src/bundle_storage.rs [hunk 1]
```diff
@@ -111,11 +111,8 @@ impl BundleStorageFileOffsets {
     }
 
     /// Return file offsets for given path.
```
Comment (guilload) [inline quickwit-storage/src/bundle_storage.rs:113]:
```suggestion
    /// Returns file offsets for given path.
```

File: quickwit-storage/src/bundle_storage.rs [hunk 2]
```diff
@@ -111,11 +111,8 @@ impl BundleStorageFileOffsets {
...
-    pub fn get(&self, path: &Path) -> StorageResult<Range<usize>> {
-        self.files
-            .get(path)
-            .cloned()
-            .ok_or_else(|| io::Error::new(io::ErrorKind::NotFound, "not found in metadata").into())
+    pub fn get(&self, path: &Path) -> Option<Range<usize>> {
+        self.files.get(path).cloned()
     }
 
     /// Return if file exists in metadata.
```
Comment (guilload) [inline quickwit-storage/src/bundle_storage.rs:118]:
```suggestion
    /// Returns whether file exists in metadata.
```

**Review by guilload — APPROVED** [review]
LGTM.
Can you add a test for `exists`, which was lacking in the original PR?

**Review by fmassot — APPROVED** [review]

File: quickwit-storage/src/bundle_storage.rs (hunk 1 above)
Comment (fulmicoton) [reply quickwit-storage/src/bundle_storage.rs:113]: Good point. I added a ticket to merge first, and improve later.

**Pull request merged**